boskos/janitor/janitor.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/client"
+)
+
+var (
+	clean    = janitorClean
+	POOLSIZE = 10 // Maximum concurrent janitor goroutines
+)
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Janitor", "http://boskos")
+	logrus.Info("Initialized boskos client!")
+	janitorPool := make(chan string)
+
+	for i := 0; i < POOLSIZE; i++ {
+		go janitor(boskos, janitorPool)
+	}
+
+	for {
+		if proj, err := boskos.Acquire("project", "dirty", "cleaning"); err != nil {
```

Comment:
Cool

An important behavior to test which is currently untested is that if:
* POOLSIZE is 10
* we have 100 things to clean up
* `janitor()` never completes

Then we will take less than 10 things off the queue.

I still do not think that is tested. BTW it looks like you can use the `len(janitorPool)` to help with this:

https://play.golang.org/
```
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Hello, playground")
	c := make(chan string, 5)
	for {
	c <- string(len(c))
	fmt.Println(len(c))
	}
}
```
Prints out:
```
Hello, playground
1
2
3
4
5
fatal error: all goroutines are asleep - deadlock!

goroutine 1 [chan send]:
main.main()
	/tmp/sandbox619724917/main.go:11 +0x140
```